libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -1131,7 +1131,23 @@ static OTA_Err_t prvProcessDataHandler( OTA_EventData_t * pxEventData )
 
             if( xErr != kOTA_Err_None )
             {
-                OTA_LOG_L2( "[%s] Failed to update job status %d\r\n", OTA_METHOD_NAME, xErr );
+                OTA_LOG_L1( "[%s] Failed to update job status %d\r\n", OTA_METHOD_NAME, xErr );
+
+                /* Job status update failed, call the platform specific code to abort the image. */
+                xErr = xOTA_Agent.xPALCallbacks.xSetPlatformImageState( xOTA_Agent.ulServerFileID, eOTA_ImageState_Aborted );
+
+                if( xErr != kOTA_Err_None )
```

Comment:
Under what conditions might `xErr != kOTA_Err_None`?  For example if this could be due to a busy/unavailable condition, should you sleep/retry here?  And what should an application do if this can never complete?  As the code is written now there's no way for the application to know that the PAL was never successfully able to set the image state to `eOTA_ImageState_Aborted`.

Reply:
For errros we can update the job status with error codes but it is not applicable here as we are in this if block when the updating job status failed. The xSetPlatformImageState function is implemented in the OTA pal, where it calls the actual platform drivers to set the image state. This can be flash location or file system depending on the platform. The OTA pal should have retries if retrying can be successful for those platform drivers. The application is abstracted from the platform state and it is handled by the ota agent by matching states of the job document and the platform before committing new image.